src/main/java/htsjdk/samtools/BamIndexValidator.java

```diff
@@ -43,49 +43,73 @@ public static int exhaustivelyTestIndex(final SamReader reader) { // throws Exce
         // look at all chunk offsets in a linear index to make sure they are valid
 
         if (reader.indexing().hasBrowseableIndex()) {
+            if (SamIndexes.BAI.fileNameSuffix.endsWith(reader.type().indexExtension())) {

```

Comment:
There's a lot of duplicated code here.  It seems like it could be deduplicated it by using AbstractBamReader and then adding an instanceof check around the linear index checking.  That's kind of gross too though...  this is probably just as good.